bitcoin/src/blockdata/constants.rs

```diff
@@ -124,7 +124,7 @@ fn bitcoin_genesis_tx(params: &Params) -> Transaction {
 pub fn genesis_block(params: impl AsRef<Params>) -> Block<Checked> {
     let params = params.as_ref();
     let transactions = vec![bitcoin_genesis_tx(params)];
-    let hash: sha256d::Hash = transactions[0].compute_txid().into();
+    let hash = sha256d::Hash::from_byte_array(transactions[0].compute_txid().to_byte_array());
```

Comment:
In e721a419a8f1131e0bc2a0891d9ba6f47d4c03e0:

Lol, we take a `Txid`, convert it to a `sha256d::Hash` and then convert it to a `TxMerkleNode`. The intermediate step really isn't needed at all, and kinda obfuscates what's happening (which is that in the single-transaction case we "cast" a txid directly into a Merkle root....this looks suspicious because it **is** suspicious! But that's Bitcoin for you.)

This is a simple enough change and a fairly small PR so I'd like you to make this change before I ack.

Reply:
To be clear: the change I'm requesting is to compress the 2 conversion lines into one.

Reply:
Oh, yes, good find! I agree it'd be better to do it in this PR. But maybe make this change in a separate preparatory commit with explanation since it was weird to begin with.

Reply:
Nice, quick touch up on my merkle root calculation knowledge and added a patch upfront that is 4 lines of red one line of green - WIN.